### PR TITLE
feat(kemono.party): add presence

### PR DESCRIPTION
**Pull request by LelIllumina**
## Description 
Added Presence for Kemono.party

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/K/Kemono.Party/presence.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+					break;
+				}
+				case "/artists": {
+					presenceData.details = "Browsing through artists";
+					break;
+				}
+				case "/artists/updated": {
+					presenceData.details = "Browsing through updated artists";
+					break;
+				}
```
Comment (DarkVillager) [inline websites/K/Kemono.Party/presence.ts:53]: Replace this with records.
E.g.
https://github.com/PreMiD/Presences/blob/7fef005cf3f93c2ae34a257bc2dc644833e5f4e6/websites/W/WebProfiles/presence.ts#L16-L26

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/K/Kemono.Party/presence.ts (hunk 1 above)
Comment (LelIllumina) [reply websites/K/Kemono.Party/presence.ts]: Ive replaced it now

Comment (DarkVillager) [reply websites/K/Kemono.Party/presence.ts]: then mark this as resolved

**Comment by RisingSunLight42** [comment]
Don't increment version number. Should be 1.0.0 when presence is added.
Also, I think your switch is really a bad usd of it, your code is really indented and can (I'm almost sure) be less indented for a better reading and maintability.

**Comment by LelIllumina** [comment]
> Don't increment version number. Should be 1.0.0 when presence is added.
> Also, I think your switch is really a bad usd of it, your code is really indented and can (I'm almost sure) be less indented for a better reading and maintability. 

Alot of the indentation was forced by Prettier and eslint, but ill try reducing it

File: websites/K/Kemono.Party/metadata.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		"id": "850319718920224798"
+	},
+	"service": "Kemono.Party",
+	"description": {
+		"en": "Welcome to Kemono, a scraper for Patreon, Pixiv Fanbox, SubscribeStar, Gumroad, Discord, DLsite, and Fantia."
+	},
+	"url": "kemono.party",
+	"regExp": "([a-z0-9-]+[.])*kemono[.]party[/]",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/mwpEkf3.png",
```
Comment (theusaf) [inline websites/K/Kemono.Party/metadata.json:14]: Logos should be [512x512px squares](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/K/Kemono.Party/metadata.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "I Am Lel#5090",
```
Comment (theusaf) [inline websites/K/Kemono.Party/metadata.json:4]:
```suggestion
		"name": "I Am Lel",
```

File: websites/K/Kemono.Party/presence.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			"/account/keys": { details: "Checking out their keys" },
+			"/posts": { details: "Browsing through posts" },
+			"/artists": { details: "Browsing through artists" },
+			"/artists/updated": { details: "Browsing through updated artists" },
+			"/favorites": { details: "Checking out their favorites" },
+			"/dmca": { details: "Reading DMCA notice" },
+		};
+
+	for (const [path, data] of Object.entries(pages))
+		if (location.pathname === path) presenceData = { ...presenceData, ...data };
```
Comment (theusaf) [inline websites/K/Kemono.Party/presence.ts:37]: `document.location` and `location` are used often in this presence. Conisder destructuring `document.location` for consistency and simplification.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/K/Kemono.Party/metadata.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+		"id": "850319718920224798"
+	},
+	"service": "Kemono.Party",
+	"description": {
+		"en": "Welcome to Kemono, a scraper for Patreon, Pixiv Fanbox, SubscribeStar, Gumroad, Discord, DLsite, and Fantia."
+	},
+	"url": "kemono.party",
+	"regExp": "([a-z0-9-]+[.])*kemono[.]party[/]",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/uLaGoCX.png",
```
Comment (DarkVillager) [inline websites/K/Kemono.Party/metadata.json:14]: The sites uses this logo:


So let's change yours.
```suggestion
	"logo": "https://i.imgur.com/jdXZ8kV.png",
```

File: websites/K/Kemono.Party/presence.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+const presence = new Presence({
+		clientId: "1067711765770682388",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/mwpEkf3.png",
```
Comment (DarkVillager) [inline websites/K/Kemono.Party/presence.ts:8]: Same as above & must be 512x512
```suggestion
		largeImageKey: "https://i.imgur.com/jdXZ8kV.png",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

File: websites/K/Kemono.Party/metadata.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+		"patreon",
+		"pixiv",
+		"fanbox",
+		"subscribestar",
+		"gumroad",
+		"discord",
+		"dlsite",
+		"fantia",
+		"anime",
+		"art"
```
Comment (Bas950) [inline websites/K/Kemono.Party/metadata.json:29]: some of these tags are probs unneeded, but I will leave that up for @EncryptedDev

**Comment by EncryptedDev** [comment]
Low quality presence, support more pages

**Pull request closed without merging**